### PR TITLE
Introduce options for sendReaction

### PR DESCRIPTION
**Pull request by ferhatelmas**
Since immutable is breaking, it's right moment to add more breaking changes such that we switch `sendReaction` similar to `sendMessage` and can have different/new options.

### CODE REVIEW

**Review by AnatolyRugalev — APPROVED** [review]
LGTM

File: src/channel.ts [hunk 1]
```diff
@@ -395,30 +395,21 @@ export class Channel<
...
+   * @param {{ enforce_unique?: boolean }} [options] Option object, {enforce_unique: true} to override any existing reaction
    *
    * @return {Promise<ReactionAPIResponse<AttachmentType, ChannelType, CommandType, MessageType, ReactionType, UserType>>} The Server Response
    */
   async sendReaction(
     messageID: string,
     reaction: Reaction<ReactionType, UserType>,
-    user_id?: string,
-    enforce_unique?: boolean,
+    options?: { enforce_unique?: boolean },
```
Comment (mahboubii) [inline src/channel.ts:405]: missing user_id

Comment (AnatolyRugalev) [reply src/channel.ts:405]: user_id is moved to reaction object

Comment (ferhatelmas) [reply src/channel.ts:405]: I did intentionally similar to `sendMessage`.

**Review by mahboubii — APPROVED** [review]

**Pull request merged**